crates/provider/src/builder.rs

```diff
@@ -112,10 +119,29 @@ pub struct ProviderBuilder<L, F, N = Ethereum> {
     network: PhantomData<fn() -> N>,
 }
 
-impl ProviderBuilder<Identity, Identity, Ethereum> {
-    /// Create a new [`ProviderBuilder`].
-    pub const fn new() -> Self {
-        Self { layer: Identity, filler: Identity, network: PhantomData }
+impl
+    ProviderBuilder<
+        Identity,
+        JoinFill<Identity, <alloy_network::Ethereum as RecommendedFillers>::RecommendedFillers>,
+        Ethereum,
+    >
+{
+    /// Create a new [`ProviderBuilder`] with the recommended filler enabled.
+    ///
+    /// Recommended fillers are preconfigured set of fillers that handle gas estimation, nonce
+    /// management, and chain-id fetching.
+    ///
+    /// Building a provider with this setting enabled will return a [`crate::fillers::FillProvider`]
+    /// with [`crate::utils::JoinedRecommendedFillers`].
+    ///
+    /// You can opt-out of using these fillers by using the `.disable_recommended_fillers()` method.
+    pub fn new() -> Self {
+        ProviderBuilder::default().with_recommended_fillers()
+    }
+
+    /// Opt-out of the recommended fillers by reseting the fillers stack in the [`ProviderBuilder`].
+    pub fn disable_recommended_fillers(self) -> ProviderBuilder<Identity, Identity, Ethereum> {
```

Comment:
doc nit: + another line saying "this is equivalent to creating the builder with ::default()"